woocommerce/woocommerce-ios
# Order Fulfillment: Pre-select the last provider used when adding a new shipment tracking

Fix #848 

This started as a simple change in the `ManualTrackingViewModel` and turned out being a new `Store`

First, the visuals:
![Screen Recording 2019-04-16 at 4 16 14 PM mov](https://user-images.githubusercontent.com/2722505/56193720-df8b7c00-6063-11e9-9857-9504c880d8be.gif)

## Changes:
### Storage:
- Added a protocol `FileStorage` and an implementation of it to read and write to a file on disk
- Added a model object `PreselectedProvider` to model the pair `siteID` and `providerName`

### Yosemite:
- Added a `AppSettingsStore` and `AppSettingsAction`. At the moment, the `Action` only exposed methods to save and retrieve a pair `ShipmentTrackingProvider` and `siteID`

### UI:
- Added a few lines to `AddTrackingViewModel` to user the `AppSettingsAction` to fetch and save the selected shipment tracking provider

## Testing
- Navigate to an order > Order fulfillment > Add Tracking
- Select a shipment tracking provider. 
- Add the tracking. or kill the app or dismiss the modal view.
- Tap Add Tracking again. The name of the provider should be populated with the last provider selected.
- Change stores, check that the name of the provider is different for different stores.

/ cc @jeanregisser 

Update release notes:
